### PR TITLE
[Issue-74] Gauss Seidel Problem and learning section

### PR DESCRIPTION
**Pull request by paddywardle**
Adding Gauss-Seidel Problem.

Files added:
- 58_gauss_seidel/solution.py: Implementation of Gauss-Seidel and test cases
- 58_gauss_seidel/learn.html: Gauss-Seidel learning section

### CODE REVIEW

File: Problems/58_gauss_seidel/learn.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    below a certain tolerance or until a maximum number of iterations is
+    reached.
+  </li>
+</ol>
+
+<h3>Matrix Form</h3>
+Gauss-Seidel can also be expressed in matrix form, in terms of the matrix
+diagonal (D), lower triangle (L) and upper triangle (U): \[ x_{i}^{(k)} = D^{-1}
+\left( b - Lx^{(k+1)} - Ux^{(k)} \right) \]
+
```
Comment (moe18) [inline Problems/58_gauss_seidel/learn.html:31]: A small example of the algo going through an iteration or 2 would be nice but not needed if you don't want to.

**Review by moe18 — COMMENTED** [review]
had a few comments, but looks good overall, let me know if you have any questions

File: Problems/58_gauss_seidel/solution.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    for _ in range(n):
+        x = gauss_seidel_it(A,b,x)
+        
+    return x
+
+def test_gauss_seidel():
+
+    # run all for 100 iterations 
+    n = 100
```
[the remark this answers is not in the document]
Comment (paddywardle) [reply Problems/58_gauss_seidel/solution.py:28]: Very good point, will sort it

File: Problems/58_gauss_seidel/solution.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    # Test case 1: basic test
+    A_1 = np.array([[4, 1, 2],[3, 5, 1],[1, 1, 3]], dtype=float)
+    b_1 = np.array([4,7,3], dtype=float)
+    expected_1 = np.linalg.solve(A_1, b_1)
```
[the remark this answers is not in the document]
Comment (paddywardle) [reply Problems/58_gauss_seidel/solution.py]: Sorted

File: Problems/58_gauss_seidel/solution.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (paddywardle) [reply Problems/58_gauss_seidel/solution.py]: Sorted

File: Problems/58_gauss_seidel/learn.html (hunk 1 above)
Comment (paddywardle) [reply Problems/58_gauss_seidel/learn.html:31]: Sorted

**Review by moe18 — APPROVED** [review]

**Pull request merged**